db.go

```diff
@@ -404,12 +410,31 @@ func (s *ColumnStore) DB(ctx context.Context, name string) (*DB, error) {
 	return db, nil
 }
 
-func (db *DB) openWAL() (WAL, error) {
-	return wal.Open(
+func (db *DB) openWAL(ctx context.Context, firstIndex uint64) (WAL, error) {
+	// persistedTables is a map from a table name to the last transaction
+	// persisted.
+	start := time.Now()
+	persistedTables := map[string]uint64{}
+	lastTx := uint64(0)
+	wal, err := wal.Open(
```

Comment:
I think this comment got lost: I find this interface pretty awkward. Thinking about it more. I think we should make `Run` a separate method so this method will look like:
```
wal, err := wal.Open
// recover, aka call Replay
db.recover()
// And finally actually start the WAL.
wal.RunAsync()
```

Reply:
hmm I dislike that API because it's brittle. You have a WAL that you can open that hasn't been replayed which seems error prone

Reply:
Also apologies for it getting lost, I thought it was resolved with your refactor into using the recover function. I had misunderstood

Reply:
No problem.

I see your point, but I feel like replaying is defined by the caller. It should be valid to `Open`, and do some operations on the wal (e.g. `Truncate`) before a `Replay`. `Replay`ing is conceptually loading wal bytes that only the caller knows how to interpret.

Additionally, this is basically the same api that `tidwall/wal` offers us. You `Open` a wal to load the state and it's ready for any action (from `Truncate`ing to `Replay`ing).

The difference with `Replay`ing at the `frostdb` user level is that they don't need to know about what is loaded and not loaded in the database and it is indeed an error to perform any operations against a database with an unloaded WAL.

Reply:
That's a good point, I'll refactor to that.

Reply:
Done